index.js

```diff
@@ -28,12 +29,24 @@ Client.prototype.get = function (url, opts, cb) {
     cb = opts
     opts = {}
   }
-  var token = this.authToken
-  if (!token) return get(url, opts, cb)
-
-  return verifyToken(this.cache, token, function (err) {
-    if (err) return cb(err)
-    return get(url, addAuthHeader(token, opts), cb)
+  var self = this
+  verifyToken(this, function (err) {
```

Comment:
I feel this could be simpler. Flow is same for all 4 methods so I will try to localize this decision tree into 1 function.